src/Angor/Client/Pages/Wallet.razor

```diff
@@ -1255,4 +1260,25 @@ else
     {
         isShowExtraWord = !isShowExtraWord;
     }
-}
+
+    public async Task<string> GetBtcValueInPreferredCurrency(decimal btcBalance)
+    {
+        return await _currencyService.GetBtcValueInPreferredCurrency(btcBalance);
+    }
+
+
+    private async Task PrecomputeFiatValues()
+    {
+        fiatValues.Clear();
+        var addresses = accountBalanceInfo.AccountInfo.AllAddresses();
+
+
+        foreach (var addressInfo in addresses)
+        {
+            var total = Money.Satoshis(addressInfo.Balance).ToUnit(MoneyUnit.BTC);
+            var fiatValue = await _currencyService.GetBtcValueInPreferredCurrency(total); // Calls the method from ICurrencyService
```

Comment:
Can't you just get the BTC value for 1 fiat and calculate the rest locally? it seems like an overkill to call the server every time for a simple multiplication

Reply:
Yeah still have i little redesign to do from when its changed to a websocket api,
generally it's supposed to be cached for 10 minutes and then if its not cached to pull it back again,
now with the websocket always having the value a different solution might be appropriate